### ISSUE TITLE
hope $this->request->get() can return body data

### ISSUE THREAD

**Issue by itbdw** [phalcon/cphalcon#13418]
### Expected and Actual Behavior

> I wanna use $this->request->get() to get body and $_REQUEST, and the getPut method does not parse json request.

> I hacked in my code, hope this can be done within Phalcon framework.

```php

$di->set('request', function () {

    class PSRequest extends \Phalcon\Http\Request {

        public function getPut(
            $name = null,
            $filters = null,
            $defaultValue = null,
            $notAllowEmpty = false,
            $noRecursive = false
        ) {
            if (strpos($this->getContentType(), 'json') !== false) {
                $put = $this->getJsonRawBody(true);
                if (gettype($put) != 'array') {
                    $put = [];
                }
                $this->_putCache = $put;
            }

            return parent::getPut($name, $filters, $defaultValue, $notAllowEmpty,
                $noRecursive);
        }

        /**
         * get data fro Body and $_REQUEST
         *
         * @param null $name
         * @param null $filters
         * @param null $defaultValue
         * @param bool $notAllowEmpty
         * @param bool $noRecursive
         * @return mixed
         */
        public function get($name = null, $filters = null, $defaultValue = null, $notAllowEmpty = false, $noRecursive = false) {
            $res = null;
            
            if (strpos($this->getContentType(), 'json') !== false) {
                $res = self::getPut($name, $filters, null, $notAllowEmpty, $noRecursive);
            }
            
            if ($res === null) {
                $res = parent::get($name, $filters, null, $notAllowEmpty, $noRecursive);
            }
            
            if ($res === null) {
                $res = $defaultValue;
            }

            return $res;
        }
    }

    return new \PSRequest();
});
```

**Comment by itbdw** [comment]
I understand your concern. Keep simple and clear.

But as we know, there are many front framework or tools (like Axios) is sending json body by default. By adding line above, retrieving data become easy again.

**Comment by Jurigag** [comment]
Yea, but there is already getPut method which is working unfortunately on raw body, we need to make it I think detect if it's json, if then parse it properly.

**Comment by itbdw** [comment]
What about getPost method? It may receive json body too when content-type is json. That's why I wrote a new getJson method to deal with this case.

So, these method should also be upgraded to compatible with json format?
```
getPut, getPost, get
hasPut, hasPost, has
```

**Comment by sergeyklay** [comment]
Fixed in the `3.4.x` branch. Feel free to open a new issue if the problem appears again. Thank you for contributing.
